src/features/HelpAndKnowledge/SolutionsHelp/SolutionDetails/Solutions/Generic/_components/AddContractorForm/index.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState } from 'react';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  Button,
+  Fieldset,
+  Form,
+  FormGroup,
+  Label,
+  TextInput
+} from '@trussworks/react-uswds';
+import {
+  MtoCommonSolutionKey,
+  useCreateMtoCommonSolutionContractorMutation
+} from 'gql/generated/graphql';
+import GetMTOSolutionContacts from 'gql/operations/ModelToOperations/GetMTOSolutionContacts';
+
+import Alert from 'components/Alert';
+import useMessage from 'hooks/useMessage';
+import useModalSolutionState from 'hooks/useModalSolutionState';
+
+type FormValues = {
+  contractorTitle: string;
+  contractorName: string;
+};
+
+const AddContractorForm = ({ closeModal }: { closeModal: () => void }) => {
+  const { t: contractorT } = useTranslation('mtoCommonSolutionContractor');
+  const { t: miscT } = useTranslation('mtoCommonSolutionContractorMisc');
+
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      contractorTitle: '',
+      contractorName: ''
+    },
+    mode: 'onChange'
+  });
+
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { isValid }
+  } = methods;
+
+  const { selectedSolution } = useModalSolutionState();
+  const { showMessage } = useMessage();
+  const [create] = useCreateMtoCommonSolutionContractorMutation({
+    refetchQueries: [
+      {
+        query: GetMTOSolutionContacts
+      }
+    ]
+  });
+  const [hasMutationError, setHasMutationError] = useState(false);
+
+  if (!selectedSolution) {
+    return null;
+  }
+
+  const onSubmit = (formData: FormValues) => {
+    create({
+      variables: {
+        key: selectedSolution.key.toUpperCase() as MtoCommonSolutionKey,
+        contractorTitle: formData.contractorTitle,
+        contractorName: formData.contractorName
+      }
+    })
```

Comment:
There's a mutation error when not populating Contractor title (which is not required).  The paradigm is not pass the property at all in the payload if it has not been populated.  We use a helper util `dirtyInput` that returns only dirty input to the payload of the mutation.  

Here we do not want to include the property `contractorTitle` in the payload.  
![Screenshot 2025-07-01 at 7 26 05 AM](https://github.com/user-attachments/assets/02914d58-4d91-4cd6-acd8-5cd33752a45c)

If `contractorTitle` was once populated and we want to remove that value, instead of passing empty string, I believe we pass `null`

Reply:
nice catch! I will update that!